src/svtk/svtk/utils/genotype_merging.py

```diff
@@ -23,41 +23,31 @@ def choose_best_genotype(sample, records):
     best_record : pysam.VariantRecord
     """
 
-    best_GT = (0, 0)
-    best_GQ = 0
-    best_record = None
-
-    if not any(['GQ' in record.samples[sample].keys() for record in records]):
-        for record in records:
-            if record.samples[sample]['GT'] != (0, 0):
-                return record
-        return records[0]
-
-    # Pick best non-reference genotype
-    # for record in records:
-    #    if record.samples[sample]['GQ'] >= best_GQ:
-        # if record is non-reference , use it
-        # or if it's a higher GQ for a reference call, use it
-    #        if record.samples[sample]['GT'] != (0, 0) or best_GT == (0, 0):
-    #            best_GT = record.samples[sample]['GT']
-    #            best_GQ = record.samples[sample]['GQ']
-    #            best_record = record
-
-    for record in records:
-        # if found non-ref GT, replace GT and GQ together
-        if record.samples[sample]['GT'] != (0, 0) and best_GT == (0, 0):
-            best_GT = record.samples[sample]['GT']
-            best_GQ = record.samples[sample]['GQ']
-            best_record = record
-        elif record.samples[sample]['GT'] == (0, 0) and best_GT != (0, 0):
-            continue
-        # if new GT  = best_GT, while found a higher GQ, replace GQ
+    # Returns true if the GT is called and non-ref
+    # TODO: does not handle multi-allelic CNVs
+    def _is_non_ref(format_fields):
+        return 'GT' in format_fields and sum(g for g in format_fields['GT'] if g is not None) > 0
```

Comment:
I don't know how long this task takes to run. I've dealt with situations where a very large data set winds up having its allele count computed. In this situation, this multi-level looping (loop over GTs, within each GT sum over loop over allele) is quite slow in python. It can be faster if you store the results in a look-up map like so:
```suggestion
_ac_map = {}
    def _is_non_ref(format_fields):
        gt = format_fields.get('GT', None)
        if gt in _ac_map:
            return _ac_map[gt] > 0
        else:
            ac = 0 if gt is None else sum(1 for a in gt if a is not None)
            _ac_map[gt] = ac
            return ac > 0
```

Reply:
Yes I've done this before for sample overlap comparisons. In this particular case, `_is_non_ref()` is only called once per record, so I don't think it saves time?